src/plugins/vis_types/timeseries/public/request_handler.ts

```diff
@@ -86,8 +85,12 @@ export const metricsRequestHandler = async ({
             .json(query.body)
             .ok({ time: query.time });
 
-          if (query.response && config.get(UI_SETTINGS.ALLOW_CHECKING_FOR_FAILED_SHARDS)) {
-            handleResponse({ body: query.body }, { rawResponse: query.response }, {}, theme);
+          if (
```

Comment:
This breaks TSVB warnings because the response is not part of the adapters. I guess you can put it in there here: https://github.com/elastic/kibana/blob/4ee7934d941b661c5fe04278eb9b8484a901cf1f/src/plugins/vis_types/timeseries/public/request_handler.ts#L86

Reply:
```
            .ok({ time: query.time, json: { rawResponse: query.response } });
```

should do the trick

Reply:
Thank you! Yes, this suggestion does ensure a warning will be shown in TSVB for shard failures:
<img width="1752" alt="image" src="https://user-images.githubusercontent.com/908371/187694374-53361c69-d3da-4f8a-b293-8da01f4f6905.png">

Hopefully we can get more functional tests of shard failure warning behavior soon! :)

Reply:
Makes sense, I can take care of this once the initial feature is in

Reply:
Created an issue for it here: https://github.com/elastic/kibana/issues/139831